source/_integrations/alarm_control_panel.markdown

```diff
@@ -10,3 +10,31 @@ ha_domain: alarm_control_panel
 
 Home Assistant can give you an interface which is similar to a classic alarm system.
 Please see [manual alarm](/integrations/manual) for alarm configuration.
+
+### Attributes
+
+Alarm system exposes the following attributes:
+
+- `code_format` - i.e., `text`, `number` or `null` if no code is needed
+- `code_arm_required` - whether code is required to control the alarm
+- `supported_features` - a bitmask representing which features are supported by the alarm. See values for each feature below. For example, if alarm supports "arm home", "arm away" and "trigger" features this value will be 11 (1 + 2 + 8).
+  - 1 - arm home
+  - 2 - arm away
+  - 4 - arm night
+  - 8 - trigger
+  - 16 - custom bypass
+  - 32 - arm vacation
```

Comment:
I think this is a bit too advanced to report. An end-user should not have to worry about internal bitmasks.

Reply:
If this is purely internal, why expose it as an attribute? But yeah, it's probably not useful for 99% of people.

Reply:
If this is internal, why expose it as an attribute? But sure, it's probably not useful for 99% of people...

Reply:
> If this is internal, why expose it as an attribute?

The frontend uses these attributes. They are hidden from the frontend towards the end user. It's a transport mechanism.